### PR TITLE
feat(docker): Support digests obtained from DockerHub

### LINKED ISSUE #24135: Include digest and releaseTimestamp from Docker Hub result
### Describe the proposed change(s).

Fetch not only the tag name but also its digest and release timestamp when using the hub.docker.com tags API

```
export interface DockerHubTags {
  next?: string;
  results: {
    name: string;
    digest: string;
    tag_last_pushed: string;
  }[];
}
```

It could be a good time to use `zod` too.

### Describe why we need/want these change(s).

Fetching digest will improve efficiency. Fetching timestamp will allow us to use minimumReleaseAge

### PR DESCRIPTION
**Pull request by zharinov**
## Context
- closes #24135

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

File: lib/modules/datasource/docker/index.ts [hunk 1]
```diff
@@ -939,7 +939,7 @@ export class DockerDatasource extends Datasource {
    * This function will filter only tags that contain a semver version
    */
   @cache({
-    namespace: 'datasource-docker-releases',
+    namespace: 'datasource-docker-releases-v2',
```
Comment (viceice) [inline lib/modules/datasource/docker/index.ts:942]: why this change? 🤔 cache will invalidate anyways after 15 min

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/datasource/docker/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/docker/index.ts:942]: Fallback cache is longer, though
